money.go

```diff
@@ -44,7 +44,12 @@ func New(cents int64, isoCode string, options ...MoneyOption) *Money {
 }
 
 func NewFromAmount(dollars float64, isoCode string, options ...MoneyOption) *Money {
-	nm := gomoney.NewFromFloat(dollars, isoCode)
+	currencyDecimals := math.Pow10(gomoney.GetCurrency(isoCode).Fraction)
+	cents := dollars * currencyDecimals
+
+	// Temp money object without value
+	money := newFromGoMoney(gomoney.New(0, isoCode), options...)
+	nm := gomoney.New(int64(money.RoundByMode(cents)), isoCode)
```

Comment:
should be `.Round` instead of `.RoundByMode`?

Reply:
updated